### DISCUSSION TITLE
Performance Development - Best setup to view vscode & angular / laravel app via subdomain with https?

### DISCUSSION THREAD

**Discussion by paulchill** [coder/code-server#5818] [Q&A]
I'm looking for some advice for an optimal setup for code-server on a LAMP stack running Laravel / Angular apps.

I have one large repo which contains the Laravel application that has a angular app inside, when i run ng build it copies and updates the laravel public directory, which i'd been using for many years 

Right now, I am currently using an old remove dev setup that:
- Views Laravel / Angular build (ng build --watch) via -> https domain pointed (route53) -> load balancer -> ec2 Apache www
- Remote SSH directly to the EC2 to write code, use terminal, composer and run ng build --watch to update the www folder while i code

What's the best way to use code server so i can write code securely via a https subdomain and view the output of the projects?

I've seen that in vscode when i run ng serve it creates https://dev.domain.com/proxy/4200/ but the bundles / resources are 404 status? is this relative path issue or something?

Can i run ng serve with a proxy **ng serve --proxy-config proxy.conf.json** to view the angular output via https://dev.domain.com/proxy/4200/ 

But how can it talk to my backend Laravel API on the same domain / server which is https://dev.domain.com/api/ ?

Instead of having the use the slow ng build that copies the code into the Laravel public folder and served via apache.

Also i'm looking to separate both the angular / laravel app and have them as separate projects and i'm thinking of keeping the back end on the production ec2 and moving the angular app to static s3 / cloud front hosting.

Effectively - what's the optimal way to code both of these app via the https subdomain in code server and then execute these each dev can view their project?

Sorry for long question touching a few subjects:)

**Comment by jsjoeio** [comment]
Hmmm...so yeah using the built-in proxy should work. I'm not experienced with Laravel so can't speak to much there. Do you have a "Hello world" repo with this setup? I'd be happy to try it out.

Ignoring the subdomain part of your question...I would do this:
- get the Angular app to work behind proxy
- get the Laravel API to work behind proxy
- connect the two
- tackle the subdomain problem

How does that sound?
